spring-integration-core/src/main/java/org/springframework/integration/store/SimpleMessageGroup.java

```diff
@@ -123,6 +128,10 @@ public int getLastReleasedMessageSequenceNumber() {
 	}
 
 	private boolean addMessage(Message<?> message) {
+		Object sequence = message.getHeaders().get(IntegrationMessageHeaderAccessor.SEQUENCE_NUMBER);
+		if (sequence instanceof Integer) {
```

Comment:
M-m-m. I think that should be fine just to request only `Integer` type for the header, like it is done right now via `messageHeaderAccessor.getSequenceNumber()`.
I mean we should just do this directly without any `if`:
```
Integer sequence = message.getHeaders().get(IntegrationMessageHeaderAccessor.SEQUENCE_NUMBER, Integer.class);
this.sequences.add(sequence != null ? sequence : 0);
```
and that's all